gcloud/storage/blob.py

```diff
@@ -109,6 +111,23 @@ def path_helper(bucket_path, blob_name):
         """
         return bucket_path + '/o/' + quote(blob_name, safe='')
 
+    @staticmethod
+    def _get_customer_encryption_headers(key):
+        """Builds customer encyrption key headers
+
+        :type key: str
+        :param key: 32 byte key to build request key and hash.
+        """
+        headers = {}
+        key_hash = base64.encodestring(hashlib.sha256(key.encode('utf-8'))
+                                       .digest()).rstrip()
```

Comment:
Don't chain the `digest()` line-to-line. It's a sign you're doing too much, use more lines.

Why are you using `base64.encodestring` instead of `b64encode` or `urlsafe_b64encode`?

Do you have a guarantee that `key` can be encoded as `utf-8`? Maybe we should add a `_from_bytes` companion to `_to_bytes`? (Already exists in `oauth2client`.)